File: Python/BinarySearch.py
"""
------------------------------------------------------------------------------------------------
USERNAME: AJgthb2002
DESCRIPTION: THIS PROGRAM RETURNS THE INDEX OF AN ELEMENT IN AN ARRAY USING BINARY SEARCH
DATE: 1/10/2021
------------------------------------------------------------------------------------------------
"""
def binary_search(array, l,h,target):
    if h>=l:
        mid= (h+l)//2
        if array[mid]==target:
            return mid
        elif array[mid]>target:
            return binary_search(array,l,mid-1,target)
        else:
            return binary_search(array,mid+1,h,target)
    else:
        return -1
    
arr=[2,3,6,8,9,10]        
print(binary_search(arr,0,len(arr)-1,9))          
  

File: C++/BinarySearch.cpp
/* 
------------------------------------------------------------------------------------------------
USERNAME: maitreyeep2
DESCRIPTION: THIS PROGRAM FINDS A NUMBER BY BINARY SEARCH,IF NUMBER IS FOUND PRINTS ITS LOCATION ELSE PRINTS "NOT FOUND".
DATE: 3/10/2021
------------------------------------------------------------------------------------------------
*/
#include <bits/stdc++.h>

using namespace std;

//SNIPPET
void bsearch()
{
	int num,no;
cout<<"how many numbers would you like to enter:";
cin>>num;
	cout<<"\n\t Enter Array ";
	 int i,a[num];
  for(i = 0;i<num;i++)
  {
      cout<<"\n\t Enter Value, in sorted manner ";
     cin>>a[i];
 }
cout<<"\n\t Enter Number to Search ";
  cin>>no;
 int min,max,mid,flag;
 min = 0;
 max = num;
 flag = 0;
 while(min<=max)
 {
     mid = (min+max)/2;
     if(no == a[mid])
	 {
	   flag=1;
	   break;
	 }
     else
	 {
	  if(no>a[mid])
	      min = mid+1;
	  else
	      max = mid-1;
	 }
 }
 if(flag == 0)
    cout<<"\n\t Not Found";
 else
 cout<<"\n\t Found at location:"<<mid;
}
int main()
{
  
//APPLICATION
 bsearch();
	return 0;
}
